### ISSUE TITLE
default-docstring-type configure is not effective in docparams checker

### ISSUE THREAD

**Issue by MrYourM** [PyCQA/pylint#4617]
### Steps to reproduce

Below is a.py

```python
class A:

    def __init__(self, name: str) -> None:
        self.name = name
     
    def a(self, cc: str) -> str:
        """[summary]

        :param cc: [description]
        :type cc: str
        :return: [description]
        :rtype: str
        """
        print(cc)
        return cc
```

and run command
```shell
pylint --disable=all --enable=parameter_documentation --default-docstring-type=google a.py
```

### Current behavior

```

------------------------------------
Your code has been rated at 10.00/10
```

### Expected behavior

I think that it should report a warning message or others

### pylint --version output

Result of `pylint --version` output:

```
2.7.2
```

**Comment by Pierre-Sassoulas** [comment]
Thank you for opening the issue. What warning would you expect here ?

**Comment by MrYourM** [comment]
> Thank you for opening the issue. What warning would you expect here ?

I want to check a google style docstring. however it did not work

**Comment by DanielNoord** [comment]
This is due to the way we handle docstrings. We try to determine a "best match" by matching the docstring against all available types. We use the `default-type` as a base and then check whether any of the other styles have a better match. If so, we take that new style as the current style for a particular docstring.

I guess what you are looking for is something like ``--enforce-docstring-style``. I guess we could consider adding it?
